applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -429,12 +429,10 @@ def _create_linear_solver(self):
             ]
             for solver_name in linear_solvers_by_speed:
                 if KratosMultiphysics.LinearSolverFactory().Has(solver_name):
-                    linear_solver_settings = KratosMultiphysics.Parameters("""{
-                        "solver_type" : \"""" + solver_name + """\"
-                    }""")
+                    linear_solver_configuration.AddEmptyValue("solver_type").SetString(solver_name)
```

Comment:
This is the only real change?, the others are only aesthetics?

Reply:
yes
Apparently you retrieve the solver_type in contact and if I don't add it this fails